graders/python/python_autograder/code_feedback.py

```diff
@@ -304,6 +304,110 @@ def bad(msg):
 
         return True
 
+    @classmethod
+    def check_dict(
+        cls,
+        name,
+        ref,
+        data,
+        partial_keys=None,
+        check_keys=False,
+        check_values=False,
+        key_type=None,
+        value_type=None,
+        accuracy_critical=False,
+        report_failure=True,
+    ):
+        """
+        Feedback.check_dict(name, ref, data)
+
+        Checks that a student dict (`data`) has all correct key-value mappings with respect to a reference dict (`ref`). It also verifies the length of keys in the student dictionary against the reference dictionary, and optionally, enforces homogeneous data types for keys (using `key_type`), values (using `value_type`), or both. Additionally, it can verify the presence of specific keys (using `partial_keys`) in the student dictionary, and can focus the comparison solely on keys (using `check_only_keys`), values (using `check_only_values`), or both.
+
+        - ``name``: Name of the dictionary that is being checked. This will be used to give feedback.
+        - ``ref``: Reference dictionary.
+        - ``data``: Student dictionary to be checked.
+        - ``partial_keys``: If not None, it takes a List of keys to check if these particular keys are present in the student's dict or not.
```

Comment:
Not a fan of the name `partial`. I would prefer `only_keys`.

Reply:
Maybe `required_keys`?

Reply:
I kept it as `only_keys`, but I would be happy to workshop the wording.